src/Hive/Controllers/UserController.cs

```diff
@@ -79,6 +145,7 @@ public UserController(HiveContext context, IProxyAuthenticationService authServi
         [ProducesResponseType(StatusCodes.Status400BadRequest)]
         public async Task<ActionResult<string>> Rename([FromQuery] string username)
         {
+            username = Uri.UnescapeDataString(username);
```

Comment:
Shouldn't this be double-checked for null before being passed to `UnescapeDataString`?